### PR TITLE
add template directory to be used for new image configs

### PR DESCRIPTION
**Pull request by imjasonh**
cc @priyawadhwa @vaikas

### CODE REVIEW

**Review by stormqueen1990 — COMMENTED** [review]
Looks like this module also needs to be ignored in `lint.sh`

File: images/TEMPLATE/config/main.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+terraform {
+  required_providers {
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+variable "extra_packages" {
+  description = "The additional packages to install (e.g. crane)."
+  default     = ["crane"]
```
Comment (priyawadhwa) [inline images/TEMPLATE/config/main.tf:9]: should this default be empty?

Comment (imjasonh) [reply images/TEMPLATE/config/main.tf:9]: yep!

File: images/TEMPLATE/main.tf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  depends_on = [module.test-latest]
+  digest_ref = module.latest.image_ref
+  tag        = "latest"
+}
+
+resource "oci_tag" "latest-dev" {
+  depends_on = [module.test-latest]
+  digest_ref = module.latest-dev.image_ref
+  tag        = "latest-dev"
+}
```
Comment (joshrwolf) [inline images/TEMPLATE/main.tf:52]: do we want to use the tagger module here?

**Comment by joshrwolf** [comment]
not to overcomplicate this, but considering how common it is for folders to contain multiple images, and how much delta there are between the two types of starting points, would this benefit from being 2 templates?

1. `TEMPLATE-single`: what this currently is
2. `TEMPLATE-multiple`: something akin to `images/vault`

or maybe:

```
images/
  TEMPLATE-SINGLE/
    this current PR
  TEMPLATE-MULTIPLE/
    a copy/pasta of something like images/vault
```

File: images/TEMPLATE/main.tf (hunk 2 above)
Comment (imjasonh) [reply images/TEMPLATE/main.tf:52]: Nope, the tagger module is moving out of the public repo, since we're only tagging `latest{-dev}` here.

**Comment by imjasonh** [comment]
> my verbiage is up for debate, I don't think we formalize this anywhere yet

I love this idea, let's save it for a followup. I mostly want to get _something_ in before the onrush of new images coming this/next week 😄

File: images/TEMPLATE/config/template.apko.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+contents:
```
Comment (vaikas) [inline images/TEMPLATE/config/template.apko.yaml:1]: Can we add comments here on how to test if you have locally built packages? In the document, I had the example, of adding the keyrings, local package dir, etc.

At least the workflow I've been using for things that include many packages that need to be built, I keep on iterating on packages, and then test them locally.

Or maybe add that in the README.md file, or does it just magically work like this? 🤔

Comment (mattmoor) [reply images/TEMPLATE/config/template.apko.yaml:1]: I think that's somewhat orthogonal to this change, but I think we should document it in `TERRAFORM.md` in the root of the repo.

File: images/TEMPLATE/tests/main.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+data "oci_exec_test" "version" {
+  digest = var.digest
+  script = "docker run --rm $IMAGE_NAME version"
+}
+
+// Invoke a script with the test.
+// TODO: Update or remove this test as appropriate.
+data "oci_exec_test" "manifest" {
+  digest = var.digest
+  script = "${path.module}/EXAMPLE_TEST.sh"
```
Comment (mattmoor) [inline images/TEMPLATE/tests/main.tf:22]: use `work_dir` instead?

File: images/TEMPLATE/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+To get started, run:
+
+```sh
+cp -R images/TEMPLATE images/[new-image-name]
+```
+
+Then, update your new `config/template.apko.yaml` file to specify packages you want the image to include, and any other necessary image config.
+
+If you need to support version streams, you can leave `packages` empty in `template.apko.yaml`, and instead add packages to the images using the `extra_packages` TF variable in `config/main.tf`.
```
Comment (mattmoor) [inline images/TEMPLATE/README.md:15]: This is missing adding the new module invocation to `main.tf` in the root of the repo (alphabetically!).

**Comment by mattmoor** [comment]
@joshrwolf I like it, but I don't love the patterns we have for multi-component images yet, so let's see what we can do to improve those first?

**Comment by joshrwolf** [comment]
> @joshrwolf I like it, but I don't love the patterns we have for multi-component images yet, so let's see what we can do to improve those first?

ack, which parts do you think need the most attention? happy to take a stab at improving them

**Review by mattmoor — APPROVED** [review]

**Pull request merged**